src/LayoutComponents/search/SearchBar.jsx

```diff
@@ -35,35 +36,30 @@ function mapDispatchToProps(dispatch) {
 }
 
 class SearchBar extends PureComponent {
-  constructor() {
-    super();
-    this.handleResults = this.handleResults.bind(this);
+  constructor(props) {
+    super(props);
+    this.handleSearchTerm = this.handleSearchTerm.bind(this);
   }
 
-  handleResults(results) {
+  handleSearchTerm(searchTerm) {
     const {
-      searchTerm,
-      updateLastPage,
-      updateSearchResults
+      updateLastPage
     } = this.props;
     const { push } = this.context.router.history;
     const { pathname } = this.context.router.history.location;
-
-    if (pathname !== '/search' && searchTerm.length >= 2) {
+    if (pathname !== '/search' && searchTerm.length > 2) {
       updateLastPage(pathname);
       push('/search');
     }
-    updateSearchResults(results);
   }
 
-
   render() {
-    const { updateSearchTerm } = this.props;
+    const { updateIsSearching, updateSearchResults } = this.props;
     return (
       <FCCSearchBar
-        handleResults={this.handleResults}
+        handleResults={updateSearchResults}
         handleSearchingStatus={updateIsSearching}
-        handleSearchTerm={updateSearchTerm}
```

Comment:
The `updateSearchTerm` action is not dispatched at all anymore. `NoResults.jsx` needs to know the search term to decide to display "you didn't search" or "no results found." I tried adding it in the `handleSearchTerm` function, and that seemed to work properly.